From: Andrew Janke <andrew@apjanke.net>
Date: Fri, 11 Sep 2020 20:01:14 -0400
Subject: Do not escape single quotes;
 only pad them with \& at beginning of lines.

Origin: backported, https://github.com/apjanke/ronn-ng/commit/e8f8baabc7ec3c48283c79a252b388df97c28ba1

They are not a control character if they are not at the beginning of a line
---
 lib/ronn/roff.rb                  |  6 ++--
 test/dots_at_line_start_test.roff |  4 +--
 test/ellipses.roff                |  2 +-
 test/entity_encoding_test.roff    |  4 +--
 test/markdown_syntax.roff         | 74 +++++++++++++++++++--------------------
 test/single_quotes.html           | 11 ++++++
 test/single_quotes.roff           |  5 +++
 test/single_quotes.ronn           |  9 +++++
 8 files changed, 71 insertions(+), 44 deletions(-)
 create mode 100644 test/single_quotes.html
 create mode 100644 test/single_quotes.roff
 create mode 100644 test/single_quotes.ronn

diff --git a/lib/ronn/roff.rb b/lib/ronn/roff.rb
index 3af434e..d9e29b8 100644
--- a/lib/ronn/roff.rb
+++ b/lib/ronn/roff.rb
@@ -339,7 +339,7 @@ module Ronn
       text.gsub!(/&#(\d+);/) { $1.to_i.chr }                # dec entities
       text.gsub!('\\', '\e')                                # backslash
       text.gsub!('...', '\|.\|.\|.')                        # ellipses
-      text.gsub!(/['.-]/) { |m| "\\#{m}" }                  # control chars
+      text.gsub!(/[.-]/) { |m| "\\#{m}" }                   # control chars
       ent.each do |key, val|
         text.gsub!(key, val)
       end
@@ -354,10 +354,12 @@ module Ronn
     # write text to output buffer
     def write(text)
       return if text.nil? || text.empty?
-      # lines cannot start with a '.'. insert zero-width character before.
+      # lines cannot start with a '.' or "'". insert zero-width character before.
       text = text.gsub(/\n\\\./, "\n\\\\&\\.")
+      text = text.gsub(/\n'/, "\n\\&\\'")
       buf_ends_in_newline = @buf.last && @buf.last[-1] == "\n"
       @buf << '\&' if text[0, 2] == '\.' && buf_ends_in_newline
+      @buf << '\&' if text[0, 1] == "'" && buf_ends_in_newline
       @buf << text
     end
 
diff --git a/test/dots_at_line_start_test.roff b/test/dots_at_line_start_test.roff
index c806100..0402d98 100644
--- a/test/dots_at_line_start_test.roff
+++ b/test/dots_at_line_start_test.roff
@@ -2,9 +2,9 @@
 .SH "NAME"
 \fBdots_at_line_start_test\fR
 .P
-There\'s a weird issue where dots at the beginning of a line generate troff warnings due to escaping\.
+There's a weird issue where dots at the beginning of a line generate troff warnings due to escaping\.
 .P
-\&\.\. let\'s see what happens\.
+\&\.\. let's see what happens\.
 .IP "" 4
 .nf
 \&\. A dot on an indented line
diff --git a/test/ellipses.roff b/test/ellipses.roff
index 01a3538..267fe1e 100644
--- a/test/ellipses.roff
+++ b/test/ellipses.roff
@@ -4,4 +4,4 @@
 .P
 Ellipses should be replaced\|\.\|\.\|\.
 .P
-\|\.\|\.\|\.also, they shouldn\'t interfere with regular dots at the beginning of a line\.
+\|\.\|\.\|\.also, they shouldn't interfere with regular dots at the beginning of a line\.
diff --git a/test/entity_encoding_test.roff b/test/entity_encoding_test.roff
index 9e5bef1..ee15a9e 100644
--- a/test/entity_encoding_test.roff
+++ b/test/entity_encoding_test.roff
@@ -12,7 +12,7 @@ Your output <i>might</i> look like this:
 .fi
 .IP "" 0
 .P
-Here\'s some special entities:
+Here's some special entities:
 .IP "\[ci]" 4
 &bull; \[ci]
 .IP "\[ci]" 4
@@ -37,4 +37,4 @@ Here\'s some special entities:
 &equiv; \(==
 .IP "" 0
 .P
-Here\'s a line that uses non\-breaking spaces to force the last\~few\~words\~to\~wrap\~together\.
+Here's a line that uses non\-breaking spaces to force the last\~few\~words\~to\~wrap\~together\.
diff --git a/test/markdown_syntax.roff b/test/markdown_syntax.roff
index 135e12c..e707c72 100644
--- a/test/markdown_syntax.roff
+++ b/test/markdown_syntax.roff
@@ -37,15 +37,15 @@ Inline markup like _italics_,  **bold**, and `code()`\.
 .SS "Philosophy"
 Markdown is intended to be as easy\-to\-read and easy\-to\-write as is feasible\.
 .P
-Readability, however, is emphasized above all else\. A Markdown\-formatted document should be publishable as\-is, as plain text, without looking like it\'s been marked up with tags or formatting instructions\. While Markdown\'s syntax has been influenced by several existing text\-to\-HTML filters \-\- including Setext \fIhttp://docutils\.sourceforge\.net/mirror/setext\.html\fR, atx \fIhttp://www\.aaronsw\.com/2002/atx/\fR, Textile \fIhttp://textism\.com/tools/textile/\fR, reStructuredText \fIhttp://docutils\.sourceforge\.net/rst\.html\fR, Grutatext \fIhttp://www\.triptico\.com/software/grutatxt\.html\fR, and EtText \fIhttp://ettext\.taint\.org/doc/\fR \-\- the single biggest source of inspiration for Markdown\'s syntax is the format of plain text email\.
+Readability, however, is emphasized above all else\. A Markdown\-formatted document should be publishable as\-is, as plain text, without looking like it's been marked up with tags or formatting instructions\. While Markdown's syntax has been influenced by several existing text\-to\-HTML filters \-\- including Setext \fIhttp://docutils\.sourceforge\.net/mirror/setext\.html\fR, atx \fIhttp://www\.aaronsw\.com/2002/atx/\fR, Textile \fIhttp://textism\.com/tools/textile/\fR, reStructuredText \fIhttp://docutils\.sourceforge\.net/rst\.html\fR, Grutatext \fIhttp://www\.triptico\.com/software/grutatxt\.html\fR, and EtText \fIhttp://ettext\.taint\.org/doc/\fR \-\- the single biggest source of inspiration for Markdown's syntax is the format of plain text email\.
 .P
-To this end, Markdown\'s syntax is comprised entirely of punctuation characters, which punctuation characters have been carefully chosen so as to look like what they mean\. E\.g\., asterisks around a word actually look like *emphasis*\. Markdown lists look like, well, lists\. Even blockquotes look like quoted passages of text, assuming you\'ve ever used email\.
+To this end, Markdown's syntax is comprised entirely of punctuation characters, which punctuation characters have been carefully chosen so as to look like what they mean\. E\.g\., asterisks around a word actually look like *emphasis*\. Markdown lists look like, well, lists\. Even blockquotes look like quoted passages of text, assuming you've ever used email\.
 .SS "Inline HTML"
-Markdown\'s syntax is intended for one purpose: to be used as a format for \fIwriting\fR for the web\.
+Markdown's syntax is intended for one purpose: to be used as a format for \fIwriting\fR for the web\.
 .P
-Markdown is not a replacement for HTML, or even close to it\. Its syntax is very small, corresponding only to a very small subset of HTML tags\. The idea is \fInot\fR to create a syntax that makes it easier to insert HTML tags\. In my opinion, HTML tags are already easy to insert\. The idea for Markdown is to make it easy to read, write, and edit prose\. HTML is a \fIpublishing\fR format; Markdown is a \fIwriting\fR format\. Thus, Markdown\'s formatting syntax only addresses issues that can be conveyed in plain text\.
+Markdown is not a replacement for HTML, or even close to it\. Its syntax is very small, corresponding only to a very small subset of HTML tags\. The idea is \fInot\fR to create a syntax that makes it easier to insert HTML tags\. In my opinion, HTML tags are already easy to insert\. The idea for Markdown is to make it easy to read, write, and edit prose\. HTML is a \fIpublishing\fR format; Markdown is a \fIwriting\fR format\. Thus, Markdown's formatting syntax only addresses issues that can be conveyed in plain text\.
 .P
-For any markup that is not covered by Markdown\'s syntax, you simply use HTML itself\. There\'s no need to preface it or delimit it to indicate that you\'re switching from Markdown to HTML; you just use the tags\.
+For any markup that is not covered by Markdown's syntax, you simply use HTML itself\. There's no need to preface it or delimit it to indicate that you're switching from Markdown to HTML; you just use the tags\.
 .P
 The only restrictions are that block\-level HTML elements \-\- e\.g\. \fB<div>\fR, \fB<table>\fR, \fB<pre>\fR, \fB<p>\fR, etc\. \-\- must be separated from surrounding content by blank lines, and the start and end tags of the block should not be indented with tabs or spaces\. Markdown is smart enough not to add extra (unwanted) \fB<p>\fR tags around HTML block\-level tags\.
 .P
@@ -64,15 +64,15 @@ This is another regular paragraph\.
 .fi
 .IP "" 0
 .P
-Note that Markdown formatting syntax is not processed within block\-level HTML tags\. E\.g\., you can\'t use Markdown\-style \fB*emphasis*\fR inside an HTML block\.
+Note that Markdown formatting syntax is not processed within block\-level HTML tags\. E\.g\., you can't use Markdown\-style \fB*emphasis*\fR inside an HTML block\.
 .P
-Span\-level HTML tags \-\- e\.g\. \fB<span>\fR, \fB<cite>\fR, or \fB<del>\fR \-\- can be used anywhere in a Markdown paragraph, list item, or header\. If you want, you can even use HTML tags instead of Markdown formatting; e\.g\. if you\'d prefer to use HTML \fB<a>\fR or \fB<img>\fR tags instead of Markdown\'s link or image syntax, go right ahead\.
+Span\-level HTML tags \-\- e\.g\. \fB<span>\fR, \fB<cite>\fR, or \fB<del>\fR \-\- can be used anywhere in a Markdown paragraph, list item, or header\. If you want, you can even use HTML tags instead of Markdown formatting; e\.g\. if you'd prefer to use HTML \fB<a>\fR or \fB<img>\fR tags instead of Markdown's link or image syntax, go right ahead\.
 .P
 Unlike block\-level HTML tags, Markdown syntax \fIis\fR processed within span\-level tags\.
 .SS "Automatic Escaping for Special Characters"
 In HTML, there are two characters that demand special treatment: \fB<\fR and \fB&\fR\. Left angle brackets are used to start tags; ampersands are used to denote HTML entities\. If you want to use them as literal characters, you must escape them as entities, e\.g\. \fB&lt;\fR, and \fB&amp;\fR\.
 .P
-Ampersands in particular are bedeviling for web writers\. If you want to write about \'AT&T\', you need to write \'\fBAT&amp;T\fR\'\. You even need to escape ampersands within URLs\. Thus, if you want to link to:
+Ampersands in particular are bedeviling for web writers\. If you want to write about 'AT&T', you need to write '\fBAT&amp;T\fR'\. You even need to escape ampersands within URLs\. Thus, if you want to link to:
 .IP "" 4
 .nf
 http://images\.google\.com/images?num=30&q=larry+bird
@@ -130,11 +130,11 @@ However, inside Markdown code spans and blocks, angle brackets and ampersands ar
 .SS "Paragraphs and Line Breaks"
 A paragraph is simply one or more consecutive lines of text, separated by one or more blank lines\. (A blank line is any line that looks like a blank line \-\- a line containing nothing but spaces or tabs is considered blank\.) Normal paragraphs should not be indented with spaces or tabs\.
 .P
-The implication of the "one or more consecutive lines of text" rule is that Markdown supports "hard\-wrapped" text paragraphs\. This differs significantly from most other text\-to\-HTML formatters (including Movable Type\'s "Convert Line Breaks" option) which translate every line break character in a paragraph into a \fB<br />\fR tag\.
+The implication of the "one or more consecutive lines of text" rule is that Markdown supports "hard\-wrapped" text paragraphs\. This differs significantly from most other text\-to\-HTML formatters (including Movable Type's "Convert Line Breaks" option) which translate every line break character in a paragraph into a \fB<br />\fR tag\.
 .P
 When you \fIdo\fR want to insert a \fB<br />\fR break tag using Markdown, you end a line with two or more spaces, then type return\.
 .P
-Yes, this takes a tad more effort to create a \fB<br />\fR, but a simplistic "every line break is a \fB<br />\fR" rule wouldn\'t work for Markdown\. Markdown\'s email\-style \fIblockquoting\fR and multi\-paragraph \fIlist items\fR work best \-\- and look better \-\- when you format them with hard breaks\.
+Yes, this takes a tad more effort to create a \fB<br />\fR, but a simplistic "every line break is a \fB<br />\fR" rule wouldn't work for Markdown\. Markdown's email\-style \fIblockquoting\fR and multi\-paragraph \fIlist items\fR work best \-\- and look better \-\- when you format them with hard breaks\.
 .SS "Headers"
 Markdown supports two styles of headers, Setext \fIhttp://docutils\.sourceforge\.net/mirror/setext\.html\fR and atx \fIhttp://www\.aaronsw\.com/2002/atx/\fR\.
 .P
@@ -149,7 +149,7 @@ This is an H2
 .fi
 .IP "" 0
 .P
-Any number of underlining \fB=\fR\'s or \fB\-\fR\'s will work\.
+Any number of underlining \fB=\fR's or \fB\-\fR's will work\.
 .P
 Atx\-style headers use 1\-6 hash characters at the start of the line, corresponding to header levels 1\-6\. For example:
 .IP "" 4
@@ -162,7 +162,7 @@ Atx\-style headers use 1\-6 hash characters at the start of the line, correspond
 .fi
 .IP "" 0
 .P
-Optionally, you may "close" atx\-style headers\. This is purely cosmetic \-\- you can use this if you think it looks better\. The closing hashes don\'t even need to match the number of hashes used to open the header\. (The number of opening hashes determines the header level\.) :
+Optionally, you may "close" atx\-style headers\. This is purely cosmetic \-\- you can use this if you think it looks better\. The closing hashes don't even need to match the number of hashes used to open the header\. (The number of opening hashes determines the header level\.) :
 .IP "" 4
 .nf
 # This is an H1 #
@@ -173,7 +173,7 @@ Optionally, you may "close" atx\-style headers\. This is purely cosmetic \-\- yo
 .fi
 .IP "" 0
 .SS "Blockquotes"
-Markdown uses email\-style \fB>\fR characters for blockquoting\. If you\'re familiar with quoting passages of text in an email message, then you know how to create a blockquote in Markdown\. It looks best if you hard wrap the text and put a \fB>\fR before every line:
+Markdown uses email\-style \fB>\fR characters for blockquoting\. If you're familiar with quoting passages of text in an email message, then you know how to create a blockquote in Markdown\. It looks best if you hard wrap the text and put a \fB>\fR before every line:
 .IP "" 4
 .nf
 > This is a blockquote with two paragraphs\. Lorem ipsum dolor sit amet,
@@ -216,7 +216,7 @@ Blockquotes can contain other Markdown elements, including headers, lists, and c
 > 1\.   This is the first list item\.
 > 2\.   This is the second list item\.
 >
-> Here\'s some example code:
+> Here's some example code:
 >
 >     return shell_exec("echo $input | $markdown_script");
 .fi
@@ -262,7 +262,7 @@ Ordered lists use numbers followed by periods:
 .fi
 .IP "" 0
 .P
-It\'s important to note that the actual numbers you use to mark the list have no effect on the HTML output Markdown produces\. The HTML Markdown produces from the above list is:
+It's important to note that the actual numbers you use to mark the list have no effect on the HTML output Markdown produces\. The HTML Markdown produces from the above list is:
 .IP "" 4
 .nf
 <ol>
@@ -291,7 +291,7 @@ or even:
 .fi
 .IP "" 0
 .P
-you\'d get the exact same HTML output\. The point is, if you want to, you can use ordinal numbers in your ordered Markdown lists, so that the numbers in your source match the numbers in your published HTML\. But if you want to be lazy, you don\'t have to\.
+you'd get the exact same HTML output\. The point is, if you want to, you can use ordinal numbers in your ordered Markdown lists, so that the numbers in your source match the numbers in your published HTML\. But if you want to be lazy, you don't have to\.
 .P
 If you do use lazy list numbering, however, you should still start the list with the number 1\. At some point in the future, Markdown may support starting ordered lists at an arbitrary number\.
 .P
@@ -308,7 +308,7 @@ To make lists look nice, you can wrap items with hanging indents:
 .fi
 .IP "" 0
 .P
-But if you want to be lazy, you don\'t have to:
+But if you want to be lazy, you don't have to:
 .IP "" 4
 .nf
 *   Lorem ipsum dolor sit amet, consectetuer adipiscing elit\.
@@ -376,7 +376,7 @@ It looks nice if you indent every line of the subsequent paragraphs, but here ag
 .nf
 *   This is a list item with two paragraphs\.
 
-    This is the second paragraph in the list item\. You\'re
+    This is the second paragraph in the list item\. You're
 only required to indent the first line\. Lorem ipsum dolor
 sit amet, consectetuer adipiscing elit\.
 
@@ -384,7 +384,7 @@ sit amet, consectetuer adipiscing elit\.
 .fi
 .IP "" 0
 .P
-To put a blockquote within a list item, the blockquote\'s \fB>\fR delimiters need to be indented:
+To put a blockquote within a list item, the blockquote's \fB>\fR delimiters need to be indented:
 .IP "" 4
 .nf
 *   A list item with a blockquote:
@@ -403,7 +403,7 @@ To put a code block within a list item, the code block needs to be indented \fIt
 .fi
 .IP "" 0
 .P
-It\'s worth noting that it\'s possible to trigger an ordered list by accident, by writing something like this:
+It's worth noting that it's possible to trigger an ordered list by accident, by writing something like this:
 .IP "" 4
 .nf
 1986\. What a great season\.
@@ -482,7 +482,7 @@ will turn into:
 .fi
 .IP "" 0
 .P
-Regular Markdown syntax is not processed within code blocks\. E\.g\., asterisks are just literal asterisks within a code block\. This means it\'s also easy to use Markdown to write about Markdown\'s own syntax\.
+Regular Markdown syntax is not processed within code blocks\. E\.g\., asterisks are just literal asterisks within a code block\. This means it's also easy to use Markdown to write about Markdown's own syntax\.
 .SS "Horizontal Rules"
 You can produce a horizontal rule tag (\fB<hr />\fR) by placing three or more hyphens, asterisks, or underscores on a line by themselves\. If you wish, you may use spaces between the hyphens or asterisks\. Each of the following lines will produce a horizontal rule:
 .IP "" 4
@@ -504,7 +504,7 @@ Markdown supports two style of links: \fIinline\fR and \fIreference\fR\.
 .P
 In both styles, the link text is delimited by [square brackets]\.
 .P
-To create an inline link, use a set of regular parentheses immediately after the link text\'s closing square bracket\. Inside the parentheses, put the URL where you want the link to point, along with an \fIoptional\fR title for the link, surrounded in quotes\. For example:
+To create an inline link, use a set of regular parentheses immediately after the link text's closing square bracket\. Inside the parentheses, put the URL where you want the link to point, along with an \fIoptional\fR title for the link, surrounded in quotes\. For example:
 .IP "" 4
 .nf
 This is [an example](http://example\.com/ "Title") inline link\.
@@ -524,7 +524,7 @@ title attribute\.</p>
 .fi
 .IP "" 0
 .P
-If you\'re referring to a local resource on the same server, you can use relative paths:
+If you're referring to a local resource on the same server, you can use relative paths:
 .IP "" 4
 .nf
 See my [About](/about/) page for details\.
@@ -569,7 +569,7 @@ The following three link definitions are equivalent:
 .IP "" 4
 .nf
 [foo]: http://example\.com/  "Optional Title Here"
-[foo]: http://example\.com/  \'Optional Title Here\'
+[foo]: http://example\.com/  'Optional Title Here'
 [foo]: http://example\.com/  (Optional Title Here)
 .fi
 .IP "" 0
@@ -631,9 +631,9 @@ And then define the link:
 .fi
 .IP "" 0
 .P
-Link definitions can be placed anywhere in your Markdown document\. I tend to put them immediately after each paragraph in which they\'re used, but if you want, you can put them all at the end of your document, sort of like footnotes\.
+Link definitions can be placed anywhere in your Markdown document\. I tend to put them immediately after each paragraph in which they're used, but if you want, you can put them all at the end of your document, sort of like footnotes\.
 .P
-Here\'s an example of reference links in action:
+Here's an example of reference links in action:
 .IP "" 4
 .nf
 I get 10 times more traffic from [Google] [1] than from
@@ -667,7 +667,7 @@ or <a href="http://search\.msn\.com/" title="MSN Search">MSN</a>\.</p>
 .fi
 .IP "" 0
 .P
-For comparison, here is the same paragraph written using Markdown\'s inline link style:
+For comparison, here is the same paragraph written using Markdown's inline link style:
 .IP "" 4
 .nf
 I get 10 times more traffic from [Google](http://google\.com/ "Google")
@@ -676,11 +676,11 @@ than from [Yahoo](http://search\.yahoo\.com/ "Yahoo Search") or
 .fi
 .IP "" 0
 .P
-The point of reference\-style links is not that they\'re easier to write\. The point is that with reference\-style links, your document source is vastly more readable\. Compare the above examples: using reference\-style links, the paragraph itself is only 81 characters long; with inline\-style links, it\'s 176 characters; and as raw HTML, it\'s 234 characters\. In the raw HTML, there\'s more markup than there is text\.
+The point of reference\-style links is not that they're easier to write\. The point is that with reference\-style links, your document source is vastly more readable\. Compare the above examples: using reference\-style links, the paragraph itself is only 81 characters long; with inline\-style links, it's 176 characters; and as raw HTML, it's 234 characters\. In the raw HTML, there's more markup than there is text\.
 .P
-With Markdown\'s reference\-style links, a source document much more closely resembles the final output, as rendered in a browser\. By allowing you to move the markup\-related metadata out of the paragraph, you can add links without interrupting the narrative flow of your prose\.
+With Markdown's reference\-style links, a source document much more closely resembles the final output, as rendered in a browser\. By allowing you to move the markup\-related metadata out of the paragraph, you can add links without interrupting the narrative flow of your prose\.
 .SS "Emphasis"
-Markdown treats asterisks (\fB*\fR) and underscores (\fB_\fR) as indicators of emphasis\. Text wrapped with one \fB*\fR or \fB_\fR will be wrapped with an HTML \fB<em>\fR tag; double \fB*\fR\'s or \fB_\fR\'s will be wrapped with an HTML \fB<strong>\fR tag\. E\.g\., this input:
+Markdown treats asterisks (\fB*\fR) and underscores (\fB_\fR) as indicators of emphasis\. Text wrapped with one \fB*\fR or \fB_\fR will be wrapped with an HTML \fB<em>\fR tag; double \fB*\fR's or \fB_\fR's will be wrapped with an HTML \fB<strong>\fR tag\. E\.g\., this input:
 .IP "" 4
 .nf
 *single asterisks*
@@ -715,7 +715,7 @@ un*frigging*believable
 .fi
 .IP "" 0
 .P
-But if you surround an \fB*\fR or \fB_\fR with spaces, it\'ll be treated as a literal asterisk or underscore\.
+But if you surround an \fB*\fR or \fB_\fR with spaces, it'll be treated as a literal asterisk or underscore\.
 .P
 To produce a literal asterisk or underscore at a position where it would otherwise be used as an emphasis delimiter, you can backslash escape it:
 .IP "" 4
@@ -773,14 +773,14 @@ will produce:
 With a code span, ampersands and angle brackets are encoded as HTML entities automatically, which makes it easy to include example HTML tags\. Markdown will turn this:
 .IP "" 4
 .nf
-Please don\'t use any `<blink>` tags\.
+Please don't use any `<blink>` tags\.
 .fi
 .IP "" 0
 .P
 into:
 .IP "" 4
 .nf
-<p>Please don\'t use any <code>&lt;blink&gt;</code> tags\.</p>
+<p>Please don't use any <code>&lt;blink&gt;</code> tags\.</p>
 .fi
 .IP "" 0
 .P
@@ -799,7 +799,7 @@ equivalent of <code>&amp;mdash;</code>\.</p>
 .fi
 .IP "" 0
 .SS "Images"
-Admittedly, it\'s fairly difficult to devise a "natural" syntax for placing images into a plain text document format\.
+Admittedly, it's fairly difficult to devise a "natural" syntax for placing images into a plain text document format\.
 .P
 Markdown uses an image syntax that is intended to resemble the syntax for links, allowing for two styles: \fIinline\fR and \fIreference\fR\.
 .P
@@ -871,9 +871,9 @@ into something like this:
 .P
 which will render in a browser as a clickable link to "address@example\.com"\.
 .P
-(This sort of entity\-encoding trick will indeed fool many, if not most, address\-harvesting bots, but it definitely won\'t fool all of them\. It\'s better than nothing, but an address published in this way will probably eventually start receiving spam\.)
+(This sort of entity\-encoding trick will indeed fool many, if not most, address\-harvesting bots, but it definitely won't fool all of them\. It's better than nothing, but an address published in this way will probably eventually start receiving spam\.)
 .SS "Backslash Escapes"
-Markdown allows you to use backslash escapes to generate literal characters which would otherwise have special meaning in Markdown\'s formatting syntax\. For example, if you wanted to surround a word with literal asterisks (instead of an HTML \fB<em>\fR tag), you can use backslashes before the asterisks, like this:
+Markdown allows you to use backslash escapes to generate literal characters which would otherwise have special meaning in Markdown's formatting syntax\. For example, if you wanted to surround a word with literal asterisks (instead of an HTML \fB<em>\fR tag), you can use backslashes before the asterisks, like this:
 .IP "" 4
 .nf
 \e*literal asterisks\e*
@@ -900,7 +900,7 @@ _   underscore
 .SH "AUTHOR"
 Markdown was created by John Gruber\.
 .P
-Manual page by Ryan Tomayko\. It\'s pretty much a direct copy of the Markdown Syntax Reference \fIhttp://daringfireball\.net/projects/markdown/syntax\fR, also by John Gruber\.
+Manual page by Ryan Tomayko\. It's pretty much a direct copy of the Markdown Syntax Reference \fIhttp://daringfireball\.net/projects/markdown/syntax\fR, also by John Gruber\.
 .SH "SEE ALSO"
 ronn(5)
 .br
diff --git a/test/single_quotes.html b/test/single_quotes.html
new file mode 100644
index 0000000..0a66ce6
--- /dev/null
+++ b/test/single_quotes.html
@@ -0,0 +1,11 @@
+<div class='mp'>
+
+<h1 id="Single-quotes">Single quotes</h1>
+<h2 id="Single-quote-at-beginning-of-line">Single quote at beginning of line</h2>
+
+<p>'This phrase should appear as normal, starting with a single quote.'</p>
+
+<h2 id="Quotes-in-middle-of-line">Quotes in middle of line</h2>
+
+<p>When you've got single quotes (') in a line, they're regular quotes.</p>
+</div>
diff --git a/test/single_quotes.roff b/test/single_quotes.roff
new file mode 100644
index 0000000..23366ef
--- /dev/null
+++ b/test/single_quotes.roff
@@ -0,0 +1,5 @@
+.TH "SINGLE_QUOTES" "" "January 1979" ""
+.SH "Single quote at beginning of line"
+\&'This phrase should appear as normal, starting with a single quote\.'
+.SH "Quotes in middle of line"
+When you've got single quotes (') in a line, they're regular quotes\.
diff --git a/test/single_quotes.ronn b/test/single_quotes.ronn
new file mode 100644
index 0000000..47a87dc
--- /dev/null
+++ b/test/single_quotes.ronn
@@ -0,0 +1,9 @@
+# Single quotes
+
+## Single quote at beginning of line
+
+'This phrase should appear as normal, starting with a single quote.'
+
+## Quotes in middle of line
+
+When you've got single quotes (') in a line, they're regular quotes.
\ No newline at end of file
